Migrated to flutter_map v5

JaffaKetchup opened the pull request:
Also fixed some of the warning/info lints that appeared.

greensopinion commented on pubspec.yaml:
```diff
@@ -1,24 +1,27 @@
-  sdk: ">=2.18.0 <4.0.0"
-  flutter: ">=2.14.0"
+  sdk: ">=3.0.0 <4.0.0"
+  flutter: ">=3.10.0"
```
are these version constraints required by `flutter_map`?

JaffaKetchup replied:
Yep, we now require these minimums.

greensopinion commented on pubspec.yaml:
```diff
@@ -1,24 +1,27 @@
   path_provider: ^2.0.2
-  vector_tile_renderer: ^4.0.0
+  vector_tile_renderer:
+    ^4.0.0
```
why a newline?

JaffaKetchup replied:
I'm not really sure tbh. I'll see if I can tidy it up later today.

JaffaKetchup replied:
Resolved in da3eb05b3e64bbd40080feb9544dffba67d76b80.
This occurred because you also had a commented out path dependency, and the auto formatter after you run 'pub upgrade' does this. I've removed the comment to put it all back on one line.

greensopinion commented on pubspec.yaml:
```diff
@@ -1,24 +1,27 @@
-  http: ^0.13.3
-  latlong2: ^0.8.0
+  http: ^1.0.0
+  latlong2: ^0.9.0
```
same per above, are these version constraints required by `flutter_map`?

JaffaKetchup replied:
Indeed, those are the minimums.

greensopinion commented on pubspec.yaml:
```diff
@@ -1,24 +1,27 @@
   async: ^2.8.2
-  executor_lib: ^1.1.0
+  executor_lib:
+    ^1.1.0
```
same?

JaffaKetchup replied:
I'm not really sure tbh. I'll see if I can tidy it up later today.

JaffaKetchup replied:
Resolved in da3eb05b3e64bbd40080feb9544dffba67d76b80.
This occurred because you also had a commented out path dependency, and the auto formatter after you run 'pub upgrade' does this. I've removed the comment to put it all back on one line.

greensopinion commented on example/lib/main.dart:
```diff
@@ -30,10 +31,10 @@ class MyHomePage extends StatefulWidget {
 }
 
-class _MyHomePageState extends State<MyHomePage> {
+class MyHomePageState extends State<MyHomePage> {
```
why public?

JaffaKetchup replied:
I'm not really sure tbh. I'll see if I can tidy it up later today.
I have a feeling my IDE may have been acting up, as I can't possibly think why I did this.

JaffaKetchup replied:
Resolved in da3eb05.

greensopinion reviewed (commented):
Thanks for the PR @JaffaKetchup ! A few questions inline below.

JaffaKetchup commented on example/lib/main.dart:
```diff
@@ -30,7 +30,7 @@ class MyHomePage extends StatefulWidget {
 
   @override
-  _MyHomePageState createState() => _MyHomePageState();
+  State<MyHomePage> createState() => _MyHomePageState();
```
This was changed because of a new lint rule which recommends avoiding exposing private members through public APIs.

JaffaKetchup commented on example/pubspec.yaml:
```diff
@@ -18,21 +18,20 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 dependencies:
   flutter:
     sdk: flutter
+  vector_tile_renderer: ^4.0.0
```
This was added because of a new lint rule which recommends directly depending on all used libraries, even if it is exposed/transitively depended on.

JaffaKetchup commented:
@greensopinion I've fixed the issues, added explanations as to why they happened, and added explanations about the replacements. Let me know if you need anything else!

Also, please keep an eye on https://github.com/fleaflet/flutter_map/projects/3. v6 will be bringing some major updates - although it is a long way off for now!

greensopinion commented:
thanks for the contribution @JaffaKetchup !

ARASHz4 commented:
Please publish in pub dev
